spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/trace/pubsub/TracePubSubAutoConfiguration.java

```diff
@@ -63,8 +62,22 @@ PubSubTracing pubSubTracing(MessagingTracing messagingTracing) {
   PublisherCustomizer tracePublisherCustomizer(PubSubTracing pubSubTracing) {
     TraceHelper helper = new TraceHelper(pubSubTracing);
 
-    return (Publisher.Builder publisherBuilder, String topic) -> {
-      publisherBuilder.setTransform(msg -> helper.instrumentMessage(msg, topic));
-    };
+    return (Publisher.Builder publisherBuilder, String topic) ->
+        publisherBuilder.setTransform(msg -> helper.instrumentMessage(msg, topic));
+  }
+
+  @Bean
+  @ConditionalOnMissingBean
+  public MessagingTracing messagingTracing(Tracing tracing) {
+    return MessagingTracing.create(tracing);
+  }
+
+  // To have the @Observed support we need to register this aspect
+  // Refers to https://spring.io/blog/2022/10/12/observability-with-spring-boot-3
+  // for more info.
+  @Bean
+  @ConditionalOnMissingBean
+  ObservedAspect observedAspect(ObservationRegistry observationRegistry) {
+    return new ObservedAspect(observationRegistry);
```

Comment:
Have we tested this anywhere?

Reply:
Yes.
Unit test [here](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/56de037878de3ab6fe4c207244cc2fc5a9451d39/spring-cloud-gcp-autoconfigure/src/test/java/com/google/cloud/spring/autoconfigure/trace/pubsub/TracePubSubAutoConfigurationTest.java#L109)

Integration test is included in the [trace sample](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/tree/fabe4e44163d8a77095322f6b72d33bf690b0025/spring-cloud-gcp-samples/spring-cloud-gcp-trace-sample).

Reply:
Thank you!